src/microbe_stage/editor/MicrobePartSelection.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Globalization;
+using Godot;
+
+/// <summary>
+///   A specialized button to display a microbe part for selection in the cell editor.
+/// </summary>
+public class MicrobePartSelection : MarginContainer
+{
+    [Export]
+    public ButtonGroup SelectionGroup;
+
+    private Label mpLabel;
+    private Button button;
+    private TextureRect iconRect;
+    private Label nameLabel;
+
+    private int mpCost;
+    private Texture partIcon;
+    private string name;
+    private bool locked;
+    private bool selected;
+
+    /// <summary>
+    ///   Emitted whenever the button is selected.
+    /// </summary>
+    [Signal]
+    public delegate void OnPartSelected(string name);
+
+    [Export]
+    public int MPCost
+    {
+        get => mpCost;
+        set
+        {
+            mpCost = value;
+            UpdateLabels();
+        }
+    }
+
+    [Export]
+    public Texture PartIcon
+    {
+        get => partIcon;
+        set
+        {
+            partIcon = value;
+            UpdateIcon();
+        }
+    }
+
+    /// <summary>
+    ///   User readable name. Note that the node Name property should be an "InternalName".
+    /// </summary>
+    [Export]
+    public string PartName
+    {
+        get => name;
+        set
+        {
+            name = value;
+            UpdateLabels();
+        }
+    }
+
+    /// <summary>
+    ///   Currently only makes the button unselectable if true.
+    /// </summary>
+    [Export]
+    public bool Locked
+    {
+        get => locked;
+        set
+        {
+            locked = value;
+
+            UpdateButton();
+            UpdateIcon();
+            UpdateLabels();
+        }
+    }
+
+    public bool Selected
+    {
+        get => selected;
+        set
+        {
+            selected = value;
+
+            UpdateButton();
+            UpdateIcon();
+        }
+    }
+
```

Comment:
I mean this class could easily have an Organelle property that is set to the organelle this is for when this is created / the editor is setup.